File: medplum-java/medplum-model/src/main/java/com/fhir/Slot.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.Boolean;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableSlot.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface Slot {
  Optional<instant> start();

  Optional<String> comment();

  Optional<Narrative> text();

  String resourceType();

  Optional<Boolean> overbooked();

  Optional<SlotStatus> status();

  Optional<List<Extension>> extension();

  Optional<code> language();

  Optional<CodeableConcept> appointmentType();

  Optional<List<ResourceList>> contained();

  Optional<id> id();

  Optional<List<Identifier>> identifier();

  Optional<List<CodeableConcept>> serviceType();

  Optional<List<Extension>> modifierExtension();

  Optional<List<CodeableConcept>> serviceCategory();

  Optional<Meta> meta();

  Optional<uri> implicitRules();

  Reference schedule();

  Optional<instant> end();

  Optional<List<CodeableConcept>> specialty();

  static ImmutableSlot.ResourceTypeBuildStage builder() {
    return ImmutableSlot.builder();
  }
}

File: medplum-typescript/packages/sdk/src/model/fhir/Slot.ts
import { code } from "./code";
import { CodeableConcept } from "./CodeableConcept";
import { Extension } from "./Extension";
import { id } from "./id";
import { Identifier } from "./Identifier";
import { instant } from "./instant";
import { Meta } from "./Meta";
import { Narrative } from "./Narrative";
import { Reference } from "./Reference";
import { ResourceList } from "./ResourceList";
import { SlotStatus } from "./SlotStatus";
import { uri } from "./uri";

export interface Slot {
    resourceType: string;
    id: id | null | undefined;
    meta: Meta | null | undefined;
    implicitRules: uri | null | undefined;
    language: code | null | undefined;
    text: Narrative | null | undefined;
    contained: ResourceList[] | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    identifier: Identifier[] | null | undefined;
    serviceCategory: CodeableConcept[] | null | undefined;
    serviceType: CodeableConcept[] | null | undefined;
    specialty: CodeableConcept[] | null | undefined;
    appointmentType: CodeableConcept | null | undefined;
    schedule: Reference;
    status: SlotStatus | null | undefined;
    start: instant | null | undefined;
    end: instant | null | undefined;
    overbooked: boolean | null | undefined;
    comment: string | null | undefined;
}
